### DISCUSSION TITLE
Prevent WebRTC stream from auto playing in browser?

### DISCUSSION THREAD

**Discussion by bobloadmire** [mrlt8/docker-wyze-bridge#690] [General]
I have a web portal setup with my webrtc streams, but when I load the webpage, it starts load all 10 streams at once which hammers my setup. Can we have a click/tap to load option somehow?

**Comment by mrlt8** [comment]
v1.11.5 _should_ respect the autoplay toggle and only start the web socket connection on click.

**Comment by bobloadmire** [comment]
ok maybe i'm an idiot, but i'm looking at the readme, only seeing patchnotes for autoplay, but no info on how to actually config it?

**Reply by mrlt8** [reply]
The auto-play toggle should be in the web-ui in the drop-down menu under the "video" button near the bottom of the page.

If you need to enable on-demand mode (where the bridge only connects to the camera when accessing a stream) you'll need to set `ON_DEMAND=True` in your docker-compose.yml.

**Reply by bobloadmire** [reply]
Oh! So i'm using the URL to view the stream directly in browser. can I append something to the URL to stop autoplay?

**Reply by bobloadmire** [reply]
for some reason video streams don't start on the UI dashboard not matter what I click, WebRTC, HLS, etc, but they do if I directly navigate to the stream WEBRTC url.

**Reply by mrlt8** [reply]
Hmm, will have to look into that. 

The port 8889 webrtc streams are being auto-generated by rtsp-simple-server so we don't have much control over that. Those pages should always connect to the camera on page load (and auto-play if the browser allows).

As for webrtc in the webUI, the auto-play should work if the bridge has an active connection to the camera - this is to prevent waking the on-demand cameras like the battery powered ones. 
You'll need to click on each camera that you want to start streaming from if the camera is on-demand and the bridge doesn't already have an active connection to that camera.

Note: Some browsers don't like auto-playing and may need a manual click/interaction to get the video to start playing.
